src/tanstack-table.tsx

```diff
@@ -100,7 +101,7 @@ type Props<T extends Table<any>> = TableProps & {
   columnFilters?: boolean;
   // labels?: Partial<typeof defaultLabels>;
   perPageOptions?: number[];
-  rowStyles?: (row: ReturnType<T["getRow"]>) => MantineStyleProp;
+  rowStyles?: (row: Row<T>, rowIndex?: number) => MantineStyleProp;
```

Comment:
I think the type here should remain `ReturnType<T["getRow"]>` here, unless that caused you problems? I think it more closely matches the type that's actually returned by the function for the end user when they add this prop

```suggestion
  rowStyles?: (row: ReturnType<T["getRow"]>, rowIndex?: number) => MantineStyleProp;
```

Reply:
it was causing me some type problems, but I can cast it (we're casting it the same way two lines earlier).